Sources/Sentry/SentryNSDataTracker.m

```diff
@@ -173,9 +207,43 @@ - (NSData *)measureNSDataFromURL:(NSURL *)url
 
     [ioSpan setDataValue:path forKey:@"file.path"];
 
+    [self mainThreadExtraInfo:ioSpan];
+
     return ioSpan;
 }
 
+- (void)mainThreadExtraInfo:(id<SentrySpan>)span
+{
+    BOOL isMainThread = [NSThread isMainThread];
+
+    [span setDataValue:@(isMainThread) forKey:@"blocked_main_thread"];
+
+    if (!isMainThread) {
+        return;
+    }
+
+    SentryThreadInspector *threadInspector = self.threadInspector;
+    SentryStacktrace *stackTrace = [threadInspector stacktraceForCurrentThreadNatively];
+
+    NSMutableArray *frames = [[NSMutableArray alloc] initWithCapacity:stackTrace.frames.count];
+    for (SentryFrame *frame in stackTrace.frames) {
+        // We dont need the frame if is a system frame
+        if ([frame.package hasPrefix:self.processInfoWrapper.processDirectoryPath]) {
+            [frames addObject:[frame serialize]];
```

Comment:
`m`: Instead of serializing the frame already here we could also add private property of `NSArray<SentryFrame *>*>` to `SentrySpan` and serialize in when calling `SentrySpan serialize`. Then also retrieving the debugImages in `extractFramesImageAddressFromSpan` would be easier cause we don't have to look into a NSDictionary and do a bunch of casting.

Reply:
Yeah, Im a bit worried about adding extra properties to SentrySpan for every new feature, but I kind liked the idea.